File: Backend JAVA/src/main/java/com/example/demo/dto/BookingResponse.java
package com.example.demo.dto;

import lombok.Data;
import java.time.LocalDate;

@Data
public class BookingResponse {
    private Long bookingId;
    private String confirmationNumber;
    private String bookingStatus;
    private String customerName;
    private String email;
    private String carName;
    private String numberPlate;
    private String pickupHub;
    private Long pickupHubId; // Added for internal logic
    private String returnHub;
    private Long carTypeId; // Added for filtering
    private LocalDate startDate;
    private LocalDate endDate;
    private Double dailyRate;
    private Double totalAmount; // Optional, if calculated
    private Double totalAddonAmount;
    private java.util.List<String> selectedAddOns;

    // Customer Details for Handover
    private String address;
    private String city;
    private String state;
    private String pincode;
    private String phoneNumber;
    private String mobileNumber;
    private String drivingLicenseNumber;
    private LocalDate dateOfBirth;
    private String passportNumber;
}

File: Backend.NET/DTOs/BookingResponse.cs
using System;
using System.Collections.Generic;

namespace FleetManagementSystem.Api.DTOs;

public class BookingResponse
{
    public long BookingId { get; set; }
    public string ConfirmationNumber { get; set; }
    public string BookingStatus { get; set; }
    public int CustomerId { get; set; }
    public string CustomerName { get; set; }
    public string Email { get; set; }
    public string CarName { get; set; }
    public string NumberPlate { get; set; }
    public string PickupHub { get; set; }
    public long? PickupHubId { get; set; }
    public string ReturnHub { get; set; }
    public long? CarTypeId { get; set; }
    public DateTime? StartDate { get; set; }
    public DateTime? EndDate { get; set; }
    public double? DailyRate { get; set; }
    public double? TotalAmount { get; set; }
    public double? TotalAddonAmount { get; set; }
    public List<string> SelectedAddOns { get; set; }
    public List<AddOnDetailDto> AddOnDetails { get; set; } // Detailed add-on info with rates
    
    // Detailed Customer Info
    public string MobileNumber { get; set; }
    public string DrivingLicenseNumber { get; set; }
    public string AddressLine1 { get; set; }
    public string City { get; set; }
    public string State { get; set; }
    public string Pincode { get; set; }
}
